Chapter 4/README.md

```diff
@@ -0,0 +1,114 @@
+# Building AI Intensive Python Applications
+
+## Chapter 6 - Embedding Models
+
+The Python script in this directory uses the `langchain-openai` library to embed textual data with OpenAI's `text-embedding-3-large` model, tailored to produce 1024 dimensional vectors versus 3072, and then the `langchain_mongodb` library to store vectors and execute vector searches.
+
+The example sets up the environment, authenticating to OpenAI with API keys, and connecting to your MongoDB Atlas cluster. Plots for three movies are then embedded and stored in MongoDB Atlas (the vector store) and different vector searches are executed to demonstrate semantic search with ranked results.
+
+### Create MongoDB Atlas cluster
+
+Follow the steps outlined in the MongoDB Atlas [documentation](https://www.mongodb.com/docs/atlas/tutorial/deploy-free-tier-cluster/#procedure) to create your cluster and enable connectivity.
+
+### Create the vector index
+
+To create an Atlas vector search index named `text_vector_index` with the provided index definition, follow these steps:
+
+1. Open your MongoDB Atlas dashboard and navigate to your project.
+2. Click on the "Database" tab in the left sidebar, then "Browse Collections".
+3. Click "Add my own data" to create an empty database and collection as follows:
+* Database name: `embeddings`
+* Collection name: `text`
+4. Once created, click on the "Create Search Index" button.
+5. Click "Atlas Vector Search / JSON Editor" and then "Next".
+6. Select the "embeddings/text" collection.
+7. Enter the Index Name as `text_vector_index`.
+8. Replace the vector index JSON definition with this:
+```json
+{ 
+  "fields": [ 
+    { 
+      "numDimensions": 1024, 
+      "path": "embedding", 
+      "similarity": "cosine", 
+      "type": "vector" 
+    } 
+  ] 
+} 
+```
+9. Click "Next" and then "Create Search Index".
+
+Now your MongoDB Atlas cluster is set up and ready to be used for storing data with vectors and executing vector searches.
+
+### Open API key and Atlas connection string
+
+Open `semantic_search.py` and add your OPEN AI API Key and MongoDB Atlas connection string as follows:
+
+```
+os.environ["OPENAI_API_KEY"] = "<your-openai-api-key>"
+ATLAS_CONNECTION_STRING = "<your-connection-string>"
```

Comment:
Perhaps mention where the connection string can be found

(under Database > Connect)